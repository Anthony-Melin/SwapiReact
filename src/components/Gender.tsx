import LabelValueFactory from "./__components__/LabelValueFactory.tsx";

export default LabelValueFactory('gender', 'Gender')
