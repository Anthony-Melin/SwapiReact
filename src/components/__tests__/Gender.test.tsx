import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Gender from '../Gender.tsx'

describe('Gender', () => {
    beforeEach(() => {
        render(<Gender value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('gender')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('gender')).toHaveTextContent('Gender: VALUE')
    })
})
