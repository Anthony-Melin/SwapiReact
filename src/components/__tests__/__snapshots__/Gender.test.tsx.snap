// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gender > should match snapshot 1`] = `
<p
  data-testid="gender"
>
  Gender
  : 
  VALUE
</p>
`;
